### ISSUE TITLE
incorrect parsing of DTS-HD MA file

### ISSUE THREAD

**Issue by karussellbremser** [audacity/audacity#3973]
### Bug description

I have a 5.1 channel PCM file that I encoded with DTS-HD Master Audio Suite 2.60.22 to DTS-HD Master Audio. I then loaded both the original PCM file and the encoded DTS-HD MA into Audacity to visually verify the waveforms.
It is normal for the DTS-HD Master Audio Suite to introduce a delay of the complete encoded track of +21ms, so I took that into account when comparing the waveforms. I verified this delay in the first part of the track, but in the second part the encoded track to my surprise had a negative delay compared to the original PCM. I found the exact spot where the delay switches.
I then reencoded the DTS-HD MA track back to PCM with UsEac3To v1.3.1 and loaded also this track into the Audacity project, and this track now had a constant delay of 21ms compared to the original PCM throughout the whole track.

In other words, Audacity decoded the DTS-HD MA track differently than eac3to. Either Audacity has a bug here, or both the DTS-HD Master Audio suite and eac3to have the same bug.

Here is a screenshot that might make the problem easier to understand (showing the center channels of each of the three tracks): https://i.imgur.com/zNzlIuY.png

### Steps to reproduce

See above. I can provide all 3 mentioned audio tracks privately.

### Expected behavior

DTS-HD MA decoding is consistent with eac3to.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

**Comment by karussellbremser** [comment]
More evidence that Audacity has a bug here: https://i.imgur.com/tc1WAgR.png
Track 1: DTS-HD MA track loaded into Audacity directly
Track 2: again DTS-HD MA reencoded to PCM with UsEac3To v1.3.1 (internally libDcaDec DTS Decoder)
Track 3: DTS-HD MA reencoded to PCM with latest ffmpeg (command "ffmpeg -i .\DTSENC.dtshd .\ffmpeg_DTSENC.w64")

If anyone provides me an email address to send a DL link to, or provide access directly to an upload server, I'll gladly share the DTS file for investigation.
